app/controllers/miq_ae_class_controller.rb

```diff
@@ -1660,6 +1661,20 @@ def x_show
     tree_select
   end
 
+  def refresh_git_domain
+    if params[:button] == "save"
+      git_based_domain_import_service.import(params[:git_repo_id], params[:git_branch_or_tag])
```

Comment:
Oh, I see what you're saying. Sorry, I thought you meant like someone else logging into their repo with different credentials or somehow accessing this method without permission. You meant someone just sends the "wrong" ID that corresponds to another repo, right? If that's the case, yeah, I believe you are correct.

Presently, I believe we treat the whole automation adding of domains/editing methods/etc as something that only a few select admins should have. That's not really an excuse to leave it the way it is, but the thought is the admins *should* be setting it up so that something like that isn't going to feasibly happen.

Also, currently the way the domain is imported is unless you pass in a tenant (which it's not doing presently), it uses the root tenant. I don't know enough about how tenants work, so I could be totally off base here, but I would assume that this root tenant has access to everything anyway, yes?

Otherwise, I agree with you, they shouldn't be able to update others' git repositories.